### ISSUE TITLE
4.1.8.a error

### ISSUE THREAD

**Issue by cdwmhcc** [bevry/version-clean#1]
``` ts
console.log(versionClean('4.1.8.a'))
// 4.1.8.
console.log(versionClean('4.1.8.beta'))
```

**Comment by balupton** [comment]
good find, these should be treated like prerelease tags right? e.g. `4.1.8-beta`. Any suggestion on what you expected for those?

---

yeah considering the readme:

```
console.log(versionClean('>= v1.0.x-prerelease < 3')) // 1.0
```

I'd expect it to just become `4.1.8`

**Comment by cdwmhcc** [comment]
> good find, these should be treated like prerelease tags right? e.g. `4.1.8-beta`. Any suggestion on what you expected for those?
> 
> yeah considering the readme:
> 
> ```
> console.log(versionClean('>= v1.0.x-prerelease < 3')) // 1.0
> ```
> 
> I'd expect it to just become `4.1.8`

Yes, 4.1.8 or 4.1.8-beta 😄

**Comment by balupton** [comment]
Cool, I'll go with `4.1.8` tomorrow.

**Comment by cdwmhcc** [comment]
> Cool, I'll go with `4.1.8` tomorrow.

``` ts
const versionCompare = (
  (prep, l, i, r) =>
  (a: string, b: string): 0 | -1 | 1 => {
    const pa = prep(a)
    const pb = prep(b)
    l = Math.max(pa.length, pb.length)
    i = 0
    r = i
    while (!r && i < l) r = ~~pa[i] - ~~pb[i++]
    return r < 0 ? -1 : r ? 1 : 0
  }
)((t: string) =>
  ('' + t)
    .replace(
      /[^\d.]+/g,
      (c) =>
        '.' +
        (c
          .replace(/[\W_]+/, '')
          .toUpperCase()
          .charCodeAt(0) -
          65536) +
        '.',
    )
    .replace(/(?:\.0+)*(\.-\d+(?:\.\d+)?)\.*$/g, '$1')
    .split('.'),
)

const list = [
  ['1.0', '1.0.0.0.0.0'],
  ['1.0', '1.0.1'],
  ['1.0b1', '1.0'],
  ['1.0b', '1.0b'],
  ['1.1', '1.0.1b'],
  ['1.1alpha', '1.1beta'],
  ['1.1rc1', '1.1beta'],
  ['1.1rc1', '1.1rc2'],
  ['1.1.0a1', '1.1a2'],
  ['1.1.0a10', '1.1.0a1'],
  ['1.1.0alpha', '1.1a'],
  ['1.1.0alpha2', '1.1b1'],
  ['1.1.0--dev', '1.1_dev'],
  ['1.0001', '1.00000.1.0.0.0.01'],
]

for (const item of list) {
  const type = {
    '-1': '<',
    '0': '=',
    '1': '>',
  }
  console.log(`${item[0]} ${type[versionCompare(item[0], item[1])]} ${item[1]}`, versionCompare(item[0], item[1]))
}
```

https://stackoverflow.com/a/53387532

**Comment by balupton** [comment]
> https://stackoverflow.com/a/53387532

for version comparison, that's outside the scope of this package, however @bevry has https://github.com/bevry/version-compare and https://github.com/bevry/version-range for that

**Comment by balupton** [comment]
Released to v1.3.0, thanks for the report!
